projects/core/src/occ/adapters/cart/converters/occ-cart-normalizer.ts

```diff
@@ -13,17 +14,26 @@ export class OccCartNormalizer implements Converter<Occ.Cart, Cart> {
 
   convert(source: Occ.Cart, target?: Cart): Cart {
     if (target === undefined) {
-      target = { ...(source as any) };
+      target = { ...(source as any) } as Cart;
     }
 
+    this.removeDuplicatePromotions(source, target);
+
+    const entriesWithPromotions = this.converter.convert(
+      target,
+      ORDER_ENTRY_PROMOTIONS_NORMALIZER
+    );
+
     if (source && source.entries) {
       target.entries = source.entries.map((entry) => ({
         ...entry,
         product: this.converter.convert(entry.product, PRODUCT_NORMALIZER),
+        promotions: entriesWithPromotions.find(
+          (item) => item.entryNumber === entry.entryNumber
+        )?.promotions,
```

Comment:
Instead of re-populating the `promotions field`, could we say on line 32:
```
target.entries = entriesWithPromotions.map((entry) => ({
```    
Or would that change the logic?  I'm not sure if the source is use don purpose after havinig removed the duplicates on the target.

Reply:
Ah but I see this is aligned wiith what is needed ini the Order normalizers.